specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.2-preview/RootCauseAnalysis.json

```diff
@@ -0,0 +1,2352 @@
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "RootCauseAnalysisDataSet"
+        ],
+        "summary": "Delete a root cause analysis data set",
+        "operationId": "DeleteRootCauseAnalysisDataSet",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "dataSetId",
+            "description": "The root cause analysis data set unique id, should be a string of uuid",
+            "required": true,
+            "type": "string",
+            "format": "uuid"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "The resource is deleted successfully or does not exist."
+          },
+          "default": {
+            "description": "Client error or server error (4xx or 5xx)",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetail"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Delete a root cause analysis data set": {
+            "$ref": "./examples/DeleteRootCauseAnalysisDataSet.json"
+          }
+        }
+      }
+    },
+    "/rootCauseAnalysis/dataSets": {
+      "get": {
+        "tags": [
+          "RootCauseAnalysisDataSet"
+        ],
+        "summary": "List root cause analysis data sets",
+        "operationId": "ListRootCauseAnalysisDataSets",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "query",
+            "name": "skip",
+            "description": "For paging, skipped number",
+            "type": "integer",
+            "format": "int32"
+          },
+          {
+            "in": "query",
+            "name": "maxpagesize",
+            "description": "The maximum number of items in one page",
+            "type": "integer",
+            "format": "int32"
+          },
+          {
+            "in": "query",
+            "name": "filter",
+            "description": "The filter expression filters out the resources to be returned, currently contains(displayName, 'filter name’) is supported",
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/RootCauseAnalysisDataSetList"
+            }
+          },
+          "default": {
+            "description": "Client error or server error (4xx or 5xx)",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetail"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List root cause analysis data sets": {
+            "$ref": "./examples/ListRootCauseAnalysisDataSets.json"
+          }
+        }
+      }
+    },
+    "/rootCauseAnalysis/dataSets/{dataSetId}/timestamps/{timestamp}/partitions/{partitionId}": {
```

Comment:
Does it mean partition will always on a single timestamp? (which means partition is not allowed to contain multiple timestamp)

Reply:
when users upload by this 'rootCauseAnalysis/dataSets/{dataSetId}/timestamps/{timestamp}/partitions/{partitionId}' API, we have a given timestamp, so user can only upload data for one timestamp. In the another POST api which contains :upload action, we can upload multiple partitions, thus can upload data for different timestamp at the same time

Reply:
What happens to the DELETE on the partitionId?

E.g. if user upload a partition spanning multiple timestamps, but call DELETE on this particular {timestamp}, what happens?

Reply:
a partition is defined by partition id and timestamp. So if we delete by timestamp, all the partitions with that timestamp will be deleted. Partitions which doesn't belong to the given timestamp will not change.

Reply:
I see. One partition is only for one timestamp. Hence it is fine to be under timestamp collection.